var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -71,10 +71,17 @@ class Gcc(AutotoolsPackage):
 
     # https://gcc.gnu.org/install/prerequisites.html
     depends_on('gmp@4.3.2:')
-    depends_on('mpfr@2.4.2:')
+    # GCC 7.3 does not compile with newer releases on some platforms, see
+    #   https://github.com/spack/spack/issues/6902#issuecomment-433030376
+    depends_on('mpfr@2.4.2:3.1.6')
```

Comment:
>> Potentially the "safest" approach is to only apply the dependency version constraint up to the latest version of GCC. That way, when someone adds a new GCC version, we can decide whether to propagate the constraint along to that version.

> Fully agree! I second to do it that way! :)

I think that an example of this approach would be a constraint like:

`depends_on('mpfr@2.4.2:3.1.6', when='@:9")`

which means that when gcc version 10 is added, that it would revert to trying newer versions of `mpfr`.